<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Hanoi</class>
 <widget class="QMainWindow" name="Hanoi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>884</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TP5: Tower of Hanoi</string>
  </property>
  <property name="styleSheet">
   <string notr="true">#Hanoi{
	background-color: #18253B;
}</string>
  </property>
  <widget class="QWidget" name="mainFrame">
   <property name="styleSheet">
    <string notr="true">#mainFrame{
	background-color: #18253B;

}
QFrame{
	border: none;
}
QLabel{
	color:  #C3C6E9;
}</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="topFrame">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>35</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>35</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QFrame{
	background-color: #18253B;
}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="spacing">
        <number>8</number>
       </property>
       <property name="leftMargin">
        <number>8</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="closeAppBtn">
         <property name="minimumSize">
          <size>
           <width>15</width>
           <height>15</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>15</width>
           <height>15</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="styleSheet">
          <string notr="true">border-radius: 7px;
background-color: #A35962;</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="minimizeAppBtn">
         <property name="minimumSize">
          <size>
           <width>15</width>
           <height>15</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>15</width>
           <height>15</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="styleSheet">
          <string notr="true">border-radius: 7px;
background-color: #EABA9A;</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="maximizeAppBtn">
         <property name="minimumSize">
          <size>
           <width>15</width>
           <height>15</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>15</width>
           <height>15</height>
          </size>
         </property>
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="styleSheet">
          <string notr="true">border-radius: 7px;
background-color: #6CD8E5;</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item alignment="Qt::AlignHCenter">
        <widget class="QLabel" name="windowTitleLabel">
         <property name="font">
          <font>
           <family>JetBrains Mono</family>
          </font>
         </property>
         <property name="text">
          <string>TP5: Tower of Hanoi Simulation Game </string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="centralFrame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>13</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QFrame" name="leftFrame">
         <property name="minimumSize">
          <size>
           <width>200</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QGroupBox" name="simulationSettingsGroupBox">
            <property name="styleSheet">
             <string notr="true">QGroupBox{
	color: white;
}</string>
            </property>
            <property name="title">
             <string>Simulation</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0" colspan="3">
              <widget class="QPushButton" name="startAutoSolveBtn">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">#startAutoSolveBtn{
	background-color:  transparent;
	border: 1px solid  #60BECE;
	border-radius: 5px;
	color:   #60BECE;
}

#startAutoSolveBtn:hover{
	background-color:  #60BECE;
	color: #18253B;
}



</string>
               </property>
               <property name="text">
                <string>Auto Solve</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0" colspan="3">
              <widget class="QPushButton" name="generateStepsBtn">
               <property name="styleSheet">
                <string notr="true">#generateStepsBtn{
	background-color:  transparent;
	border: 1px solid  #A587A3;
	border-radius: 5px;
	color:   #A587A3;
}

#generateStepsBtn:hover{
	background-color:  #A587A3;
	color: #18253B;
}




</string>
               </property>
               <property name="text">
                <string>Generate Steps</string>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QPushButton" name="resetGameBtn">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">#resetGameBtn{
	background-color:  transparent;
	border: 1px solid  #9C5D66;
	border-radius: 5px;
	color:   #9C5D66;
}

#resetGameBtn:hover{
	background-color:  #9C5D66;
	color: #18253B;
}




</string>
               </property>
               <property name="text">
                <string>Reset</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="undoMoveBtn">
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">#undoMoveBtn{
	background-color:  transparent;
	border: 1px solid  #EABA9A;
	border-radius: 5px;	
	color:   #EABA9A;
}

#undoMoveBtn:hover{
	background-color:  #EABA9A;
	color: #18253B;
}




</string>
               </property>
               <property name="text">
                <string>Undo</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="configGroupBox">
            <property name="styleSheet">
             <string notr="true">QGroupBox{
	color: white;
}</string>
            </property>
            <property name="title">
             <string>Config</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="1" column="0">
              <widget class="QLabel" name="disckCountLabel">
               <property name="text">
                <string>Disk Count:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSlider" name="deltaTimeSlider">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The delta time between each step executed by the AutoSolver.&lt;/p&gt;&lt;p&gt;min: 100ms&lt;/p&gt;&lt;p&gt;max: 1000ms&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="styleSheet">
                <string notr="true">QSlider::groove:horizontal {
    border-radius: 5px;
    height: 10px;
    margin: 0px;
    background-color: rgb(52, 59, 72);
}
QSlider::groove:horizontal:hover {
    background-color: rgb(55, 62, 76);
}
QSlider::handle:horizontal {
    background-color: #82B2C2;
    border: none;
    height: 10px;
    width: 10px;
    margin: 0px;
    border-radius: 5px;
}
QSlider::handle:horizontal:hover {
    background-color: rgb(195, 155, 255);
}
QSlider::handle:horizontal:pressed {
    background-color: rgb(255, 121, 198);
}</string>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox">
               <property name="styleSheet">
                <string notr="true">QSpinBox{
	background-color: #303948;
	color: #848992;
}
</string>
               </property>
              </widget>
             </item>
             <item row="6" column="0">
              <widget class="QLabel" name="verboseLabel">
               <property name="text">
                <string>Sound Effect:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="lockedLabel">
               <property name="text">
                <string>Locked:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1" alignment="Qt::AlignRight">
              <widget class="QCheckBox" name="lockedCheckBox">
               <property name="styleSheet">
                <string notr="true">QCheckBox::indicator {
    border: 3px solid rgb(52, 59, 72);
    width: 15px;
    height: 15px;
    border-radius: 10px;
    background: rgb(44, 49, 60);
}

QCheckBox::indicator:hover {
    border: 3px solid rgb(58, 66, 81);
}

QCheckBox::indicator:checked {
    background: 3px solid #95535E;
   border: 3px solid rgb(52, 59, 72);  
}</string>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="intervalLabel">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The delta time between each step executed by the AutoSolver.&lt;/p&gt;&lt;p&gt;min: 100ms&lt;/p&gt;&lt;p&gt;max: 1000ms&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="text">
                <string>Interval Δt:</string>
               </property>
              </widget>
             </item>
             <item row="6" column="1" alignment="Qt::AlignRight">
              <widget class="QCheckBox" name="sfxCheckBox">
               <property name="styleSheet">
                <string notr="true">QCheckBox::indicator {
    border: 3px solid rgb(52, 59, 72);
    width: 15px;
    height: 15px;
    border-radius: 10px;
    background: rgb(44, 49, 60);
}

QCheckBox::indicator:hover {
    border: 3px solid rgb(58, 66, 81);
}

QCheckBox::indicator:checked {
    background: 3px solid #8C7088;
   border: 3px solid rgb(52, 59, 72);  
}</string>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="stepsGroupBox">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>130</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">QGroupBox{
	color: white;
}

QScrollBar:vertical {
    border: none;
    background: rgb(52, 59, 72);
    width: 8px;
    margin: 21px 0 21px 0;
    border-radius: 0px;
 }
 QScrollBar::handle:vertical {  
    background: #739EB0;
    min-height: 25px;
    border-radius: 4px
 }
 QScrollBar::add-line:vertical {
     border: none;
    background: rgb(55, 63, 77);
     height: 20px;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
     subcontrol-position: bottom;
     subcontrol-origin: margin;
 }
 QScrollBar::sub-line:vertical {
    border: none;
    background: rgb(55, 63, 77);
    height: 20px;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
    subcontrol-position: top;
    subcontrol-origin: margin;
 }
 QScrollBar::up-arrow:vertical, QScrollBar::down-arrow:vertical {
     background: none;
 }

 QScrollBar::add-page:vertical, QScrollBar::sub-page:vertical {
     background: none;
 }
</string>
            </property>
            <property name="title">
             <string>Steps</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_9">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <property name="spacing">
              <number>0</number>
             </property>
             <item row="0" column="0" alignment="Qt::AlignHCenter">
              <widget class="QListWidget" name="stepsViewer">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>100</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>100</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">QListWidget{
	background-color: transparent;
	color: #A4B3D8;
}
QListWidget::item:selected {
	background: #8F7591;
}</string>
               </property>
               <property name="itemAlignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="movesInfoGroupBox">
            <property name="styleSheet">
             <string notr="true">QGroupBox{
	color: white;
}</string>
            </property>
            <property name="title">
             <string>Move Count</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <widget class="QLCDNumber" name="moveCountLCD">
               <property name="styleSheet">
                <string notr="true">color: white;
</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="contentFrame">
         <property name="styleSheet">
          <string notr="true">#contentFrame{
	background-color: #1B2C4A;
	border: 1px solid #7A93AE;
}</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="fromPoleBtn">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>450</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton:hover{
	border-bottom:  10px solid grey;
	border-top: none;
	border-left: none;
	border-right: none;
}

QPushButton:pressed{
	border-bottom:  10px solid #98B088;
	border-top: none;
	border-left: none;
	border-right: none;
	background-color: none;
}</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
            <property name="default">
             <bool>false</bool>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="auxPoleBtn">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>450</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton:hover{
	border-bottom:  10px solid grey;
	border-left: none;
	border-top: none;
	border-right: none;
}

QPushButton:pressed{
	border-bottom:  10px solid #98B088;
	border-left: none;
	border-top: none;
	border-right: none;
	background-color: none;
}</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="toPoleBtn">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>450</height>
             </size>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton:hover{
	border-bottom:  10px solid grey;
	border-left: none;
	border-top: none;
	border-right: none;
}

QPushButton:pressed{
	border-bottom:  10px solid #98B088;
	border-left: none;
	border-top: none;
	border-right: none;
	background-color: none;
}</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="bottomFrame">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>30</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>20</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QFrame{
	background-color: #18253B;
}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>9</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item alignment="Qt::AlignTop">
        <widget class="QLabel" name="creditsLabel">
         <property name="font">
          <font>
           <family>JetBrains Mono</family>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>by Ouzaher Saad</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="action_New">
   <property name="text">
    <string>&amp;New</string>
   </property>
  </action>
  <action name="action_Undo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="action_Undo_All">
   <property name="text">
    <string>Undo &amp;All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionAuto_Play">
   <property name="text">
    <string>Auto &amp;Play</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>&amp;2</string>
   </property>
  </action>
  <action name="action_3">
   <property name="text">
    <string>&amp;3</string>
   </property>
  </action>
  <action name="action_4">
   <property name="text">
    <string>&amp;4</string>
   </property>
  </action>
  <action name="action_5">
   <property name="text">
    <string>&amp;5</string>
   </property>
  </action>
  <action name="action_6">
   <property name="text">
    <string>&amp;6</string>
   </property>
  </action>
  <action name="action_7">
   <property name="text">
    <string>&amp;7</string>
   </property>
  </action>
  <action name="action_8">
   <property name="text">
    <string>&amp;8</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
